components/ui-api-layer/internal/domain/k8s/pod_converter.go

```diff
@@ -0,0 +1,187 @@
+package k8s
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kyma-project/kyma/components/ui-api-layer/internal/gqlschema"
+	v1 "k8s.io/api/core/v1"
+)
+
+type podConverter struct{}
+
+func (c *podConverter) ToGQL(in *v1.Pod) (*gqlschema.Pod, error) {
+	if in == nil {
+		return nil, nil
+	}
+
+	containerStates := c.containerStatusesToGQLContainerStates(in.Status.ContainerStatuses)
```

Comment:
Maybe would be better if all logic for extracting statuses would be in separate package, eg. `k8s/status/pod.go`. What do you think about that @michal-hudy?

Reply:
Sure, I agree 😄